### ISSUE TITLE
Array removal + edit does not cause a conflict

### ISSUE THREAD

**Issue by caofeilong** [marcello3d/trimerge#27]
```
const s1 = { f: [{ id: '122', name: '123' }] }
      const s2 = { f: [] }
      const s3 = { f: [{ id: '122', name: '90999' }] }
      const merger = combineMergers(trimergeJsonDeepEqual, trimergeJsonObject, trimergeArrayCreator((item, index, path) => {
        return item.id || index
      }, true))
      let mm = function (origin, s1, s2, path) {
        let mdata = merger(origin, s1, s2, path, mm)
        if (mdata instanceof CannotMergeClass) {
          return 'ddd'
        }
        return mdata
      }
      let da = merger(s1, s2, s3, ['data'], mm)
      console.info(da)
```

The f attribute in s1, s2 and s3 in the above code should be a conflicting relationship, and the final result of the merge should be {f:[' DDD ']} but the result from my side is {f:[]}.

**Comment by marcello3d** [comment]
Good find! The way things are implemented now, the keys are compared first before looking at the values. I'll have to think a little bit about how to support this scenario.

**Comment by marcello3d** [comment]
Added some unit tests in #19

**Comment by marcello3d** [comment]
@caofeilong this should be fixed in the latest release (1.3.0-alpha.14)
